src/url.js

```diff
@@ -111,10 +116,15 @@ export function parseUrlDeprecated(url, opt_nocache) {
  * @param {!HTMLAnchorElement} a
  * @param {string} url
  * @param {LruCache=} opt_cache
+ *   Cache is altogether ignored on ESM builds, since an <a> is not used.
  * @return {!Location}
  * @restricted
  */
 export function parseUrlWithA(a, url, opt_cache) {
+  if (IS_ESM) {
+    return /** @type {!Location} */ (new URL(url));
```

Comment:
The `a` anchor here is used because it can be from _another_ window context with it's own base HREF. We need to specify the `base` param, and ensure the `a` is using the window's base:

```suggestion
    a.href = '';
    return /** @type {!Location} */ (new URL(url, a.href));
```

Reply:
I'm now setting `a.href = ''` when the element is created, PTAL.